numpy/lib/_shape_base_impl.py

```diff
@@ -1260,17 +1274,30 @@ def tile(A, reps):
         tup = tuple(reps)
     except TypeError:
         tup = (reps,)
-    d = len(tup)
     if all(x == 1 for x in tup) and isinstance(A, _nx.ndarray):
         # Fixes the problem that the function does not make a copy if A is a
         # numpy array and the repetitions are 1 in all dimensions
-        return _nx.array(A, copy=True, subok=True, ndmin=d)
+        return _nx.array(A, copy=True, subok=True, ndmin=len(tup))
     else:
         # Note that no copy of zero-sized arrays is made. However since they
         # have no data there is no risk of an inadvertent overwrite.
-        c = _nx.array(A, copy=False, subok=True, ndmin=d)
-    if (d < c.ndim):
-        tup = (1,)*(c.ndim-d) + tup
+        c = _nx.array(A, copy=False, subok=True, ndmin=len(tup))
+
+    if isinstance(axis, int):
+        if not isinstance(reps, int):
+            raise ValueError("`reps` should be an integer")
+        axis, reps = (axis,), (reps,)
```

Comment:
This assumes `reps` is an integer, but it could have been passes as a tuple. (The docstring for reps suggests this as it states "array_like")

Reply:
I've updated the docstring for `reps` to include int.

I included this check because `np.tile(A, (2, 2), axis=0)` is self contradictory.